frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1509,491 +1779,248 @@ function ReaderFooter:addToMainMenu(menu_items)
-                    {
-                        text = _("Thick"),
-                        checked_func = function()
-                            return not self.settings.progress_style_thin
-                        end,
-                        callback = function()
-                            self.settings.progress_style_thin = nil
-                            local bar_height = self.settings.progress_style_thick_height
-                            self.progress_bar:updateStyle(true, bar_height)
-                            self:setTocMarkers()
-                            self:refreshFooter(true, true)
-                        end,
-                    },
-                    {
-                        text = _("Thin"),
-                        checked_func = function()
-                            return self.settings.progress_style_thin
-                        end,
-                        callback = function()
-                            self.settings.progress_style_thin = true
-                            local bar_height = self.settings.progress_style_thin_height
-                            self.progress_bar:updateStyle(false, bar_height)
-                            self:refreshFooter(true, true)
+                            return T(_("Book-title item: %1".. "%"), self.settings.book_title_max_width_pct)
                         end,
-                    },
-                    {
-                        text = _("Set size"),
-                        callback = function()
-                            local value, value_min, value_max, default_value
-                            if self.settings.progress_style_thin then
-                                default_value = PROGRESS_BAR_STYLE_THIN_DEFAULT_HEIGHT
-                                value = self.settings.progress_style_thin_height or default_value
-                                value_min = 1
-                                value_max = 12
-                            else
-                                default_value = PROGRESS_BAR_STYLE_THICK_DEFAULT_HEIGHT
-                                value = self.settings.progress_style_thick_height or default_value
-                                value_min = 5
-                                value_max = 28
-                            end
+                        callback = function(touchmenu_instance)
+                            local SpinWidget = require("ui/widget/spinwidget")
+                            local items = SpinWidget:new{
+                                value = self.settings.book_title_max_width_pct,
+                                value_min = 10,
+                                value_step = 5,
+                                value_hold_step = 20,
+                                value_max = 100,
+                                unit = "%",
+                                title_text = _("Book-title item"),
+                                info_text = _("Maximum percentage of screen width used for book-title"),
```

Comment:
We can add default value to the spinwidget.
https://github.com/koreader/koreader/blob/6eabb0b94386116f6edf0cf0a47ae67334e70751/frontend/apps/reader/modules/readerfooter.lua#L470
(Didn't check other spinwidgets for the same change)